modmail.py

```diff
@@ -1,154 +1,70 @@
-        return
-
-    # Get member object
-    reaction_user = payload.member
-
-    # Ignore if bot
-    if reaction_user.bot:
-        return
-
-    # Get unicode emoji
-    emoji = payload.emoji.name
-
-    # Get message object
-    message = await modmail_channel.fetch_message(payload.message_id)
-
-    await handle_reaction(emoji, message, reaction_user)
-
-async def handle_reaction(emoji, message, reaction_user):
-    """Handles reactions for ModMail embeds.
-
-    Args:
-        emoji (discord.PartialEmoji): The emoji being used.
-        message (discord.Message): The current message.
-        reaction_user (discord.Member): The user who triggered the reaction event.
-    """
-    ticket = db.get_ticket_by_message(message.id)
-    if ticket['ticket_id'] == -1:
-        return
-
-    await message.remove_reaction(emoji, reaction_user)
-
-    embed_actions = embed_reactions.embed_reactions(bot, guild, modmail_channel, reaction_user, ticket)
-
-    if str(emoji) == '🗣️':
-        await embed_actions.message_reply()    
-    elif str(emoji) == '❎':
-        await embed_actions.message_close()    
-    elif str(emoji) == '⏲️':
-        ticket_user = await bot.fetch_user(ticket['user'])
-        await embed_actions.message_timeout(ticket_user)    
-        
-@bot.event
-async def on_ready():
-    global guild, modmail_channel
-
-    guild = bot.get_guild(config['guild'])
-
-    if guild is None:
-        print('Failed to find Guild from provided ID.')
-        await bot.close()
-        return
-
-    modmail_channel = guild.get_channel(config['channel'])
-
-    if modmail_channel is None:
-        print('Failed to find Modmail Channel from provided ID.')    
-        await bot.close()
-        return
-      
-    await bot.change_presence(activity=discord.Game(name=config['status']), status=discord.Status.online)
-
-    bot.add_cog(Listeners(bot, guild, modmail_channel))
-    bot.add_cog(CommandActions(bot, modmail_channel))
-
-def ready():
-    if db.init():
-        print('Database sucessfully initialized!')
-    else:
-        print('Error while initializing database!')
-        return False
-
-    if "guild" not in config:
-        print('No Guild ID provided.')
-        return False
-
-    if "channel" not in config:
-        print('No Channel ID provided.')
-        return False
-    
-    if "prefix" not in config:
-        print('Failed to find prefix in config.')
-        return False
-
-    return True
-
-success = ready()
-
-if success:    
-    bot.run(config['token']) 
+        for cog in ('commands', 'listeners'):
+            try:
+                await bot.load_extension(f'cogs.{cog}')
+                logger.debug(f'Imported cog "{cog}".')
+            except commands.errors.NoEntryPointError as e:
+                logger.warning(e)
+            except commands.errors.ExtensionNotFound as e:
+                logger.warning(e)
+            except commands.errors.ExtensionFailed as e:
+                logger.error(e)
```

Comment:
It might be worth exiting entirely if these cogs do not load? Can the bot function at all without them?

I noticed that the setup hooks include system exits, but they might be better placed here since all the components are necessary for this bot from my understanding.